### PR TITLE
Feature/update device info

### PR DESCRIPTION
**Pull request by Mee-gu**
update device info

### CODE REVIEW

File: cocos/renderer/backend/DeviceInfo.h [hunk 1]
```diff
@@ -6,20 +6,43 @@
 
 CC_BACKEND_BEGIN
 
+enum class FeaturesInfo : uint32_t
```
Comment (minggo) [inline cocos/renderer/backend/DeviceInfo.h]: FeaturesInfo -> FeatureType?

File: cocos/renderer/backend/metal/DeviceInfoMTL.h [hunk 2]
```diff
@@ -12,11 +12,14 @@ class DeviceInfoMTL : public DeviceInfo
 	virtual ~DeviceInfoMTL() = default;
 
     virtual bool init() override;
+    virtual const char* getVendor() const override;
+    virtual const char* getRenderer() const override;
+    virtual const char* getVersion() const override;
+    virtual const char* getExtension() const override;
+    virtual bool checkForFeatureSupported(const FeaturesInfo& feature) override;
```
Comment (minggo) [inline cocos/renderer/backend/metal/DeviceInfoMTL.h]: FeaturesInfo is an enum class, don't have to use reference.

File: cocos/renderer/backend/metal/DeviceInfoMTL.mm [hunk 3]
```diff
@@ -1,29 +1,71 @@
...
     _maxAttributes = 31;
+    _maxSamplesAllowed = 16;
+    
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_IOS)
+    _maxTextureUnits = 31;
+     _maxTextureSize = 4096;
+#else
+    _maxTextureUnits = 128;
+    _maxTextureSize = 16384;
+#endif
```
Comment (minggo) [inline cocos/renderer/backend/metal/DeviceInfoMTL.mm:356]: Hard code?  Is there any way to get these informations?

File: tests/cpp-tests/Classes/AppDelegate.cpp [hunk 4]
```diff
@@ -56,18 +56,18 @@ void AppDelegate::initGLContextAttrs()
...
     auto glview = director->getOpenGLView();
     if(!glview) {
         glview = GLViewImpl::create("Cpp Tests");   
         director->setOpenGLView(glview);
     }
+    
+    // As an example, load config file
+    // FIXME:: This should be loaded before the Director is initialized,
+    // FIXME:: but at this point, the director is already initialized
+    Configuration::getInstance()->loadConfigFile("configs/config-example.plist");
```
Comment (minggo) [inline tests/cpp-tests/Classes/AppDelegate.cpp]: Why move the codes here?

Comment (Mee-gu) [reply tests/cpp-tests/Classes/AppDelegate.cpp]: yep, it's no need to move the codes.

**Comment by Mee-gu** [comment]
update review

**Pull request merged**